sysl2/sysl/ints_view.go

```diff
@@ -0,0 +1,555 @@
+		_, okB := v.mod.Apps[appB].Endpoints[epB].Attrs[restrictBy]
+		if _, ok := v.mod.Apps[appA].Attrs[restrictBy]; !ok && restrictBy != "" {
+			if _, ok := v.mod.Apps[appB].Attrs[restrictBy]; !ok {
+				continue
+			}
+		}
+		if !okA && restrictBy != "" && !okB {
+			continue
+		}
+		clusters[appA] = append(clusters[appA], epA)
+		if appA != appB && !v.mod.Apps[appA].Endpoints[epA].IsPubsub {
+			clusters[appA] = append(clusters[appA], epB+" client")
+		}
+		clusters[appB] = append(clusters[appB], epB)
+	}
+
+	keys := []string{}
+	for k := range clusters {
+		keys = append(keys, k)
+	}
+
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		v.VarManagerForTopState(k)
+		strSet := MakeStrSet(clusters[k]...)
+		for _, m := range strSet.ToSortedSlice() {
+			v.VarManagerForState(k + " : " + m)
+		}
+		v.stringBuilder.WriteString("}\n")
+	}
+}
+
+func (v *IntsDiagramVisitor) buildClusterForComponentView(apps []string) map[string]string {
+	nameMap := map[string]string{}
+	clusters := map[string][]string{}
+	for _, v := range apps {
+		cluster := strings.Split(v, " :: ")
+		if len(cluster) > 1 {
+			clusters[cluster[0]] = append(clusters[cluster[0]], v)
+		}
+	}
+
+	for k, v := range clusters {
+		if len(v) <= 1 {
+			delete(clusters, k)
+		}
+		for _, s := range v {
+			nameMap[s] = strings.Split(s, " :: ")[1]
+		}
+	}
+
+	for k, apps := range clusters {
+		fmt.Fprintf(v.stringBuilder, "package \"%s\" {\n", k)
+		for _, n := range apps {
+			v.VarManagerForComponent(n, nameMap)
+		}
+		v.stringBuilder.WriteString("}\n")
+	}
+
+	return nameMap
+}
+
+func (v *IntsDiagramVisitor) generateStateView(viewParams viewParams, params *IntsParam) string {
+	v.stringBuilder.WriteString("@startuml\n")
+	if viewParams.diagramTitle != "" {
+		fmt.Fprintf(v.stringBuilder, "title %s\n", viewParams.diagramTitle)
+	}
+	v.stringBuilder.WriteString(StateStart)
+	if viewParams.highLightColor != "" {
+		fmt.Fprintf(v.stringBuilder, "  BackgroundColor<<highlight>> %s\n", viewParams.highLightColor)
+	}
+	if viewParams.arrowColor != "" {
+		fmt.Fprintf(v.stringBuilder, "  ArrowColor %s\n", viewParams.arrowColor)
+	}
+
+	if viewParams.indirectArrowColor != "" && viewParams.indirectArrowColor != ArrowColorNone {
+		fmt.Fprintf(v.stringBuilder, "  ArrowColor<<indirect>> %s\n", viewParams.indirectArrowColor)
+	}
+	v.stringBuilder.WriteString("}\n")
+	v.buildClusterForStateView(params.integrations, viewParams.restrictBy)
+	var processed []string
+	for _, dep := range params.integrations {
+		appA := dep.Self.Name
+		appB := dep.Target.Name
+		epA := dep.Self.Endpoint
+		epB := dep.Target.Endpoint
+		_, restrictByAppA := v.mod.Apps[appA].Attrs[viewParams.restrictBy]
+		_, restrictByAppB := v.mod.Apps[appB].Attrs[viewParams.restrictBy]
+		_, restrictByEpA := v.mod.Apps[appA].Endpoints[epA].Attrs[viewParams.restrictBy]
+		_, restrictByEpB := v.mod.Apps[appB].Endpoints[epB].Attrs[viewParams.restrictBy]
+		if viewParams.restrictBy != "" && !restrictByAppA && !restrictByAppB {
```

Comment:
this does not match python functionality, python version checks if restrict_by is in any of the apps or endpoints. id represented by restrict_by is required to be in both.
My sample below should make this clear.

Reply:
foo_id is only needed on systema and one of its endpoint.

```
IntegratedSystem:
    integrated_endpoint_1:
        Systema <- endpoint
    integrated_endpoint_2:
        Systemb <- endpoint

Systema [foo_id="y"]:
    endpoint [foo_id="y"]:
        Systemb <- endpoint2

    endpoint_2 [foo_id="y"]:
        Systemb <- endpoint2

Project [appfmt="%(appname)", epfmt="%(epname)"]:
    _ [restrict_by="foo_id"]:
        IntegratedSystem

Systemb:
    endpoint: ...
    endpoint2: ...
```